### ISSUE TITLE
[Bug] [Rule: S4328] Rule gets triggered when using path maps

### ISSUE THREAD

**Issue by LautaroPetaccio** [SonarSource/SonarTS#677]
**I want to report a bug.**

**SonarTS version:** 1.7
**Node.js version:** 8.11.3
**TypeScript version:** 2.9.2
**SonarQube version:** 6.7
**SonarLint version and used IDE:** Not using it.
**TSLint version:** Not running as a TSLint extension.

**Rule key:** S4328
The one named here as "Dependencies should be explicit"

**Reproducer** 
When scanning a project with *path aliases* or *path maps*, described in the tsconfig.json as:
```
path: {
  "helpers/*": ["src/helpers/*"]
}
```
The S4328 rule gets triggered.

**Code example:**
```
import clipboard from "helpers/clipboard";
```
Results in:
```
(MAJOR) Module 'helpers' is not listed as dependency in package.json rule
```

**Logs**
Don't have.

**Expected behavior**
This rule shouldn't get triggered by this.

**Comment by vilchik-elena** [comment]
I can't find `path` option for tsconfig. Do you mean `paths`?

**Comment by LautaroPetaccio** [comment]
Oh, yes, it's paths, sorry.
We are using that feature to make the paths more manageable and we're gettings tons of false positives.
Also, thanks for replying.

**Comment by vilchik-elena** [comment]
This rule comes from TSLint. I've created an issue on their side: palantir/tslint#4083

**Comment by chris5287** [comment]
FYI this has been fixed in tslint: https://github.com/palantir/tslint/pull/3979

Hopefully we'll get a fix here soon as well :)

**Comment by premuditha** [comment]
Hi @vilchik-elena I'm using SonarTS 1.8.0.3332 (typescript) in SonarCloud and I have added my paths as follows in tsconfig.json

```
"paths": {
      "app/*": ["app/*"],
      "models/*": ["app/models/*"],
      "modules/*": ["app/modules/*"],
      "services/*": ["app/services/*"],
      "shared/*": ["app/shared/*"],
      "testing/*": ["app/shared/testing/*"]
}
```

and `no-implicit-dependencies` rule setting as follows in `tslint.json` to skip checking following modules in package.json.

```
"no-implicit-dependencies": [
      true,
      "dev",
      [
        "app",
        "models",
        "services",
        "testing",
        "shared",
        "modules",
        "environments"
      ]
    ]
```

everything works well when I run` ng lint` command locally and Im not getting any Dependencies should be explicit warnings. 

Bug in SonarCloud, I'm still getting errors like **Module 'shared' is not listed as dependency in package.json**.

I'm just wondering how to use the `whitelist` parameter and pass modules that should skip this rule.

**Comment by vilchik-elena** [comment]
You should configure parameter of rule S4328 on SonarCloud, SonarTS doesn't rely at all on your tslint config

**Comment by promontis** [comment]
@vilchik-elena when we use absolute paths for an import (say src/app/module) instead of relative paths, this seems to trigger this rule as well... isn't that strange? Or shouldn't we be using absolute paths? 

I've also tried to add a path:

```
 "paths": {
      "src/*": ["src/*"]
    }
```
But it still says that src isn't `Module 'src' is not listed as dependency in package.json`. 
How to fix this?

**Comment by vilchik-elena** [comment]
@promontis Please create a new issue using our template

**Comment by rafiek** [comment]
@vilchik-elena How to use the `whitelist` parameter and pass modules that should skip this rule.

**Comment by vilchik-elena** [comment]
@rafiek In SonarQube UI open the rule page and edit value of parameter

**Comment by rafiek** [comment]
@vilchik-elena There is no way to edit the value of the parameter. We are running Community EditionVersion 7.2.1 (build 14109).

**Comment by vilchik-elena** [comment]
@rafiek do you have admin rights?

**Comment by aliakseikrauchanka** [comment]
Sorry @vilchik-elena @promontis is there a way to specify wildcard for **whitelist** parameter? 
We have couple of packages starting with some prefix and want to skip all of them in **whitelist**.
Thanks in advance!

**Comment by vilchik-elena** [comment]
@aliakseikrauchanka Nope

**Comment by MlleBeuze** [comment]
Hi @vilchik-elena,
We got Code Smells saying the same thing.
We tried to specify the tsconfig.json in the configuration in the "sonar.typescript.tsconfigPath" emplacement, as you can see on the attached file. Our tsconfig.json file is at the root /tsconfig.json and it's where we defined all the alias paths.

But when we run SonarQube again, Code Smells are still there. "Module '@shared' is not listed as dependency in package.json." Where are we wrong?

Thank you in advance for your help.
![image](https://user-images.githubusercontent.com/13280573/51323053-6d58ca80-1a67-11e9-8353-008d8ee239da.png)

**Comment by vilchik-elena** [comment]
@MlleBeuze please create a new issue using template

**Comment by mikaoelitiana** [comment]
@MlleBeuze have you solved this issue? I am facing it but didn't find the resolution.

**Comment by Ovsdrak** [comment]
<img width="1460" alt="Captura de Pantalla 2020-05-11 a la(s) 19 20 55" src="https://user-images.githubusercontent.com/20526969/81624917-10e88f80-93bd-11ea-8d07-d9b7442f9612.png">

**Comment by MlleBeuze** [comment]
@mikaoelitiana Hey sorry, I don't think we solved it, but you can find the issue here: https://github.com/SonarSource/SonarTS/issues/803#issuecomment-459712982
Some people replied, maybe it can help... Good luck !

**Comment by MarkWilsoncom** [comment]
Having problems with this too. Pops up as a major issue on Sonarqube.
Will this be fixed? It´s been known for 2+ years.

**Comment by saberduck** [comment]
hello @MarkWilsoncom , this repository is going to be archived. TypeScript analysis is done with sonar JavaScript analyzer. I would recommend to open the thread on our community forum here https://community.sonarsource.com/ and provide details which version of SonarQube do you use, code sample raising the issue, etc...
